mobile_base_controller:
  type        : "diff_drive_controller/DiffDriveController"
  left_wheel  : 'left_wheel_joint'
  right_wheel : 'right_wheel_joint'
  # FIXME
  pose_covariance_diagonal : [0.001, 0.001, 1000000.0, 1000000.0, 1000000.0, 1000.0]
  twist_covariance_diagonal: [0.001, 0.001, 1000000.0, 1000000.0, 1000000.0, 1000.0]
  # Using STL meshes so cannot read automatically
  wheel_separation : 0.0511
  wheel_radius : 0.084
  # Some limits to prevent abrupt acc/dec and topple-over
  linear:
    x:
      has_acceleration_limits: true
      max_acceleration       : 10 # m/s^2
      min_acceleration       : -10 # m/s^2

arm_controller:
  type: position_controllers/JointTrajectoryController
  joints:
    - arm_joint_1
    - arm_joint_2
    - arm_joint_3
    - hand_joint

joint_state_controller:
  type: joint_state_controller/JointStateController
  publish_rate: 50

# https://answers.ros.org/question/157261/manipulator-shakeswobbles/
gazebo_ros_control:
  pid_gains:
    arm_joint_1: {p: 100.0, d: 0.0, i: 0.0, i_clamp: 0.0, i_clamp_min: 0.0, i_clamp_max: 0.0}
    arm_joint_2: {p: 100.0, d: 0.0, i: 0.0, i_clamp: 0.0, i_clamp_min: 0.0, i_clamp_max: 0.0}
    arm_joint_3: {p: 100.0, d: 0.0, i: 0.0, i_clamp: 0.0, i_clamp_min: 0.0, i_clamp_max: 0.0}
    hand_joint:  {p: 100.0, d: 0.0, i: 0.0, i_clamp: 0.0, i_clamp_min: 0.0, i_clamp_max: 0.0}

